solr/core/src/java/org/apache/solr/handler/component/HighlightComponent.java

```diff
@@ -184,6 +185,20 @@ public void process(ResponseBuilder rb) throws IOException {
     }
   }
 
+  /**
+   * Normalizes parameters between highlighters
+   */
+  private SolrParams normalizeParameters(SolrParams params) {
```

Comment:
You've coded this such that SIMPLE_PRE overrides TAG_PRE which is not what we want I think?  Furthermore, this is coded such that it only overrides at the global level which won't work for field-specific settings like `f.myfieldname.hl.tag.pre` which we'd want to examine `f.myfieldname.hl.simple.pre`.  I appreciate where you were going with this, but in light of the latter point, I think you should simply modify the Solr UH adapter to lookup say "pre" like so:

    String preTag = params.getFieldParam(fieldName, HighlightParams.TAG_PRE, 
           params.getFieldParam(fieldName, HighlightParams.SIMPLE_PRE, "<em>");
     );